src/probnum/_config.py

```diff
@@ -0,0 +1,83 @@
+import contextlib
+from typing import Any
+
+
+class Configuration:
+    """
+    Configuration over which some mechanics of probnum can be controlled dynamically.
+
+    ``probnum`` provides some configurations together with default values. These
+    are listed in the tables below.
+    Additionally, the user can register own configuration entries via the method
+    :meth:`register`. Configuration entries can only be registered once and can only
+    be used (accessed or overwritten) once they have been registered.
+
+    - ``probnum.randvars``
+
+    +----------------------------------+-----------------------------------------------+
+    | Config entry                     | Description                                   |
+    +==================================+===============================================+
+    | ``covariance_inversion_damping`` | A (typically small) value that is per         |
+    |                                  | default added to the diagonal of covariance   |
+    |                                  | matrices in order to make inversion           |
+    |                                  | numerically stable.                           |
+    +----------------------------------+-----------------------------------------------+
+    | ``...``                          | ...                                           |
+    +----------------------------------+-----------------------------------------------+
+
+    - ``probnum.diffeq``
+
+    - ``...``
```

Comment:
what about this? Is it needed?

Reply:
It's a stub for config options yet to come.

Reply:
Sure, but I think with the existing `randvars` entry right above this stub, it does not really serve a purpose. Or do I misunderstand?

Reply:
Only if we want to have module-wide configs and also document them. Then, the intention is to have a list of tables for the possible configs.
But -- again -- I see your point, in that I also find that debatable, and would -- at least for now -- agree to implement probnum-wide (global) configs only.

Reply:
Another option might be to add the tables to the landing pages of the corresponding subpackages

Reply:
Please keep on discussing, but my comment was much more minor than that :D I just wanted to point out that the line 28 serves no purpose in terms of "you would put the info here", because an example is already given with randvars :)

Reply:
That's no example. That's the documentation. The bullet point `diffeq` will then also be filled with a respective table full of config options and their descriptions.

Reply:
ah okay. My bad.